### PR TITLE
MRG: Use environment.yml in CIs

### PR DESCRIPTION
**Pull request by larsoner**
This should be nearly equivalent to what we already do, just with fewer lines and using an `environment.yml` file. This moves us toward more easily testing things like #4523 when we think Py3k might actually be ready for primetime.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,85 +8,66 @@ cache:
...
 matrix:
     include:
+        # No data + style testing
+        - os: linux
+          env: DEPS=nodata MNE_DONTWRITE_HOME=true MNE_FORCE_SERIAL=true MNE_SKIP_NETWORK_TEST=1
+               CONDA_DEPENDENCIES="numpy scipy matplotlib sphinx nose pytest pytest-cov"
+               PIP_DEPENDENCIES="flake8 numpydoc codespell git+git://github.com/PyCQA/pydocstyle.git codecov check-manifest pytest-sugar"
+               OPTION="--doctest-ignore-import-errors"
+
         # Full (Linux, 2.7) first half
```
Comment (agramfort) [inline .travis.yml]: remove first half as you removed the comment about second half

File: .travis.yml [hunk 2]
```diff
@@ -8,85 +8,66 @@ cache:
...
+               SPLIT=0
         - os: linux
-          env: CONDA_DEPENDENCIES="numpy scipy matplotlib pandas scikit-learn h5py pillow statsmodels mayavi nose pytest pytest-cov"
-               PIP_DEPENDENCIES="git+git://github.com/nipy/PySurfer.git nitime faulthandler joblib nibabel codecov pytest-sugar pytest-faulthandler"
-               MNE_DIRS="minimum_norm preprocessing realtime simulation stats time_frequency viz"
-               OPTION=""
+          env: CONDA_ENVIRONMENT="environment.yml"
+               SPLIT=1
 
         # OSX first half
```
Comment (agramfort) [inline .travis.yml]: idem about first half

File: environment.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+- pandas
+- h5py
+- statsmodels
+- nose
+- pytest
+- pytest-cov
+- joblib
+- jupyter
+- sphinx
+- spyder
```
Comment (agramfort) [inline environment.yml]: I agree that spyder is probably a good idea for users but maybe an overkill for testing?

Comment (larsoner) [reply environment.yml]: I don't think it adds much download time. And in the meantime this sets us up to properly test the purpose of this (set up a usable environment for users that gets tested by CIs)

**Comment by agramfort** [comment]
looks like travis did not run

**Comment by agramfort** [comment]
is this now working?

but only on 2.7 right?

**Comment by larsoner** [comment]
Yes 2.7, OSX and Linux, probably not on CircleCI (some weird error), and AppVeyor also causes crashes. But it's some progress at least. I'll revert CircleCI so we can merge and iterate further.

File: doc/manual/channel_interpolation.rst [hunk 4]
```diff
@@ -58,7 +58,7 @@ where :math:`G_{ds} \in R^{M \times N}` computes :math:`g_{m}(\boldsymbol{r_i},
 
 To interpolate bad channels, one can simply do:
 
-	>>> evoked.interpolate_bads(reset_bads=False)
+	>>> evoked.interpolate_bads(reset_bads=False)  # doctest: +SKIP
```
Comment (larsoner) [inline doc/manual/channel_interpolation.rst]: FYI apparently our doctests aren't really running on CircleCI currently, we'll need to fix it separately though

**Pull request merged**